torchft/checkpointing/transport_test.py

```diff
@@ -0,0 +1,147 @@
+import threading
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from datetime import timedelta
+from typing import Callable, Dict, List
+from unittest import TestCase
+
+import torch
+import torch.distributed as dist
+from torch.distributed.tensor import DeviceMesh, DTensor, distribute_tensor
+
+from torchft.checkpointing.transport import CheckpointTransport
+
+TIMEOUT_REGEX = r"(Timed out|timed out|timeout|time out)"
+
+
+def assertStateDictEqual(
+    self: TestCase, a: Dict[str, object], b: Dict[str, object]
+) -> None:
+    for k, v1 in a.items():
+        v2 = b[k]
+        if isinstance(v1, DTensor) and isinstance(v2, DTensor):
+            torch.testing.assert_close(v1._local_tensor, v2._local_tensor)
+            self.assertEqual(v1._spec, v2._spec)
+        elif isinstance(v1, torch.Tensor) and isinstance(v2, torch.Tensor):
+            torch.testing.assert_close(v1.cpu(), v2.cpu())
+        else:
+            self.assertEqual(v1, v2)
+
+
+def run_multi_recovery_test(
+    self: TestCase,
+    init_transport: Callable[[int, int], CheckpointTransport[Dict[str, object]]],
+    device: torch.device,
+) -> None:
+    """
+    This runs multi node recovery tests for a given transport function.
+
+    This tests up to 3 nodes.
+    """
+    WORLD_SIZE: int = 3
```

Comment:
nit: should this be a parameter instead? The function def implies that it's configurable

Reply:
This test explicitly tests sending at sizes 2/3 and will error out if this changes, I can update the comment